File: app/src/java/java/com/example/ama/viewer/di/modules/AppModule.java
package com.example.ama.viewer.di.modules;

import android.content.Context;

import com.example.ama.viewer.ViewerApp;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module(includes = {
        UtilsModule.class,
        FragmentModule.class,
        PresenterModule.class,
        DbModule.class,
        NetworkModule.class,
        RepositoryModule.class})
public class AppModule {

    @Provides
    @Singleton
    Context provideContext(ViewerApp app) {
        return app;
    }
}

File: app/src/kotlin/java/com/example/ama/viewer/di/modules/AppModule.kt
package com.example.ama.viewer.di.modules

import android.content.Context
import com.example.ama.viewer.ViewerApp
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module(includes = [
    UtilsModule::class,
    FragmentModule::class,
    PresenterModule::class,
    RepositoryModule::class,
    NetworkModule::class,
    DbModule::class])
class AppModule {

    @Provides
    @Singleton
    fun provideContext(app: ViewerApp): Context = app
}